### PR TITLE
Profile screen and Header component

### PR DESCRIPTION
**Pull request by VitorDaynno**
- [Tela de edição de usuário](https://trello.com/c/025vRzPl/9-tela-de-edi%C3%A7%C3%A3o-de-usu%C3%A1rio)

### CODE REVIEW

File: src/style/header.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import {StyleSheet} from 'react-native';
+
+export default StyleSheet.create({
+  header: {
+    backgroundColor: '#828489',
+    borderBottomColor: '#272932',
+    borderBottomWidth: 1,
+    flexDirection: 'row',
+    height: '7%',
+    width: '100%',
```
Comment (VitorDaynno) [inline src/style/header.js:10]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

**Comment by VitorDaynno** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 100
           

Clones added
============
- src/components/home.js  1
- src/pages/home.js  1
- src/components/profile.js  2
- src/style/profile.js  2
- src/pages/profile.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182&bid=17780242)

File: src/style/profile.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    color: '#191919',
+  },
+  email: {
+    alignSelf: 'center',
+    marginBottom: '45%',
+    fontSize: 20,
+  },
+  removeAccount: {
+    flexDirection: 'row',
+    backgroundColor: '#c70039',
```
Comment (VitorDaynno) [inline src/style/profile.js:57]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/header.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    super(props);
+  }
+
+  goBack = () => {
+    const {destiny} = this.props;
+    destiny();
+  };
+
+  render() {
+    const {name, isMenu} = this.props;
```
Comment (VitorDaynno) [inline src/components/header.js:18]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/profile.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    alignSelf: 'center',
+    marginTop: '13%',
+    flexDirection: 'column',
+    justifyContent: 'space-around',
+    backgroundColor: 'white',
+  },
+  containerPhoto: {
+    borderColor: '#020122',
+    borderWidth: 2,
+    width: '55%',
```
Comment (VitorDaynno) [inline src/style/profile.js:28]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/profile.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
+import {StyleSheet} from 'react-native';
```
Comment (VitorDaynno) [inline src/style/profile.js:1]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/profile.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    justifyContent: 'center',
+    alignItems: 'center',
+    marginTop: 22,
+  },
+  modalView: {
+    margin: 20,
+    backgroundColor: 'white',
+    borderRadius: 20,
+    padding: 35,
+    alignItems: 'center',
```
Comment (VitorDaynno) [inline src/style/profile.js:84]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/home.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  },
+  photo: {
+    color: '#222831',
+  },
+  body: {
+    height: '84%',
+    backgroundColor: 'white',
+    borderRadius: 15,
+    margin: 10,
+    borderColor: '#222831',
```
Comment (VitorDaynno) [inline src/style/home.js:30]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/pages/profile.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import React, {Component} from 'react';
+import {View} from 'react-native';
```
Comment (VitorDaynno) [inline src/pages/profile.js:2]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/header.js [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import {StyleSheet} from 'react-native';
+
+export default StyleSheet.create({
+  header: {
+    backgroundColor: '#828489',
+    borderBottomColor: '#272932',
```
Comment (VitorDaynno) [inline src/style/header.js:6]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/profile.js [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+    this.setState({avatar, email, nickname});
+  }
+
+  removeAccount = async () => {
+    try {
+      const token = await AsyncStorage.getItem('@token');
+      await api.delete('/v1/user', {
+        headers: `Bearer ${token}`,
+      });
+      Actions.Login();
```
Comment (VitorDaynno) [inline src/components/profile.js:45]: Codacy Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/profile.js [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+    fontWeight: 'bold',
+    color: '#191919',
+  },
+  email: {
+    alignSelf: 'center',
+    marginBottom: '45%',
+    fontSize: 20,
+  },
+  removeAccount: {
+    flexDirection: 'row',
```
Comment (VitorDaynno) [inline src/style/profile.js:56]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/page.js [hunk 12]
```diff
@@ -0,0 +1,7 @@
+import {StyleSheet} from 'react-native';
+
+export default StyleSheet.create({
+  page: {
+    height: '100%',
```
Comment (VitorDaynno) [inline src/style/page.js:5]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/home.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    marginRight: 10,
+    margin: 5,
+    borderRadius: 100,
+  },
+  photo: {
+    color: '#222831',
+  },
+  body: {
+    height: '84%',
+    backgroundColor: 'white',
```
Comment (VitorDaynno) [inline src/style/home.js:27]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/profile.js [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+      } else {
+        this.setState({
+          errorMessage: I18n.t('genericError'),
+          modalVisible: true,
+        });
+      }
+    }
+  };
+
+  setModalVisible = modalVisible => {
```
Comment (VitorDaynno) [inline src/components/profile.js:63]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/pages/home.js [hunk 15]
```diff
@@ -0,0 +1,18 @@
+import React, {Component} from 'react';
```
Comment (VitorDaynno) [inline src/pages/home.js:1]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/home.js [hunk 16]
```diff
@@ -1,12 +1,42 @@
 import React, {Component} from 'react';
-import {View} from 'react-native';
+import {View, Text, TouchableOpacity} from 'react-native';
+import Icon from 'react-native-vector-icons/SimpleLineIcons';
+import {Actions} from 'react-native-router-flux';
+import AsyncStorage from '@react-native-community/async-storage';
+
+import Styles from '../style/home';
```
Comment (VitorDaynno) [inline src/components/home.js:7]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/profile.js [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+
+export default StyleSheet.create({
+  avatar: {
+    alignSelf: 'center',
+    marginTop: 30,
+  },
+  userDetails: {
+    flex: 1,
+    flexDirection: 'column',
+    backgroundColor: '#dbdbdb',
```
Comment (VitorDaynno) [inline src/style/profile.js:11]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/profile.js [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+    width: '100%',
+    height: '114%',
+    marginTop: 26,
+  },
+  name: {
+    marginTop: 10,
+    fontSize: 40,
+    alignSelf: 'center',
+    fontWeight: 'bold',
+    color: '#191919',
```
Comment (VitorDaynno) [inline src/style/profile.js:48]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/header.js [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+
+import Styles from '../style/header';
+
+export default class Header extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  goBack = () => {
+    const {destiny} = this.props;
```
Comment (VitorDaynno) [inline src/components/header.js:13]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/profile.js [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+    backgroundColor: '#2196F3',
+  },
+  textStyle: {
+    color: 'white',
+    fontWeight: 'bold',
+    textAlign: 'center',
+  },
+  modalText: {
+    marginBottom: 15,
+    textAlign: 'center',
```
Comment (VitorDaynno) [inline src/style/profile.js:107]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/profile.js [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+  containerPhoto: {
+    borderColor: '#020122',
+    borderWidth: 2,
+    width: '55%',
+    height: '35%',
+    borderRadius: 23,
+    alignSelf: 'center',
+    marginTop: '13%',
+    flexDirection: 'column',
+    justifyContent: 'space-around',
```
Comment (VitorDaynno) [inline src/style/profile.js:34]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/header.js [hunk 22]
```diff
@@ -0,0 +1,36 @@
+import React, {Component} from 'react';
+import {View, Text} from 'react-native';
```
Comment (VitorDaynno) [inline src/components/header.js:2]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/style/profile.js [hunk 23]
```diff
@@ -0,0 +1,109 @@
...
+  },
+  centeredView: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+    marginTop: 22,
+  },
+  modalView: {
+    margin: 20,
+    backgroundColor: 'white',
```
Comment (VitorDaynno) [inline src/style/profile.js:81]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/profile.js [hunk 24]
```diff
@@ -0,0 +1,136 @@
...
+  Text,
+  TouchableOpacity,
+  Image,
+  Modal,
+  TouchableHighlight,
+} from 'react-native';
+import Icon from 'react-native-vector-icons/SimpleLineIcons';
+import AsyncStorage from '@react-native-community/async-storage';
+import {Actions} from 'react-native-router-flux';
+import ImagePicker from 'react-native-image-picker';
```
Comment (VitorDaynno) [inline src/components/profile.js:13]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/header.js [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+import Icon from 'react-native-vector-icons/SimpleLineIcons';
+
+import Styles from '../style/header';
```
Comment (VitorDaynno) [inline src/components/header.js:5]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

**Comment by VitorDaynno** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 7
           

Clones added
============
- src/pages/profile.js  1
- src/components/profile.js  2
- src/pages/home.js  1
- src/style/profile.js  2
- src/components/home.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182&bid=17780242)

File: src/components/header.js (hunk 3 above)
Comment (VitorDaynno) [inline src/components/header.js:18]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/home.js [hunk 26]
```diff
@@ -1,12 +1,42 @@
...
+    this.setState({avatar, email, nickname});
   }
 
   render() {
-    return <View />;
+    const {nickname} = this.state;
+    return (
+      <View style={[Styles.page]}>
+        <View style={[Styles.header]}>
+          <TouchableOpacity onPress={() => Actions.Profile()}>
```
Comment (VitorDaynno) [inline src/components/home.js:32]: Codacy Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/profile.js (hunk 10 above)
Comment (VitorDaynno) [inline src/components/profile.js:45]: Codacy Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/profile.js [hunk 27]
```diff
@@ -0,0 +1,136 @@
...
+  changeAvatar = () => {
+    const options = {
+      title: 'Select Avatar',
+      storageOptions: {
+        skipBackup: true,
+        path: 'images',
+      },
+    };
+
+    ImagePicker.showImagePicker(options, response => {
```
Comment (VitorDaynno) [inline src/components/profile.js:76]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/profile.js [hunk 28]
```diff
@@ -0,0 +1,136 @@
...
+      if (response.error) {
+        console.log('ImagePicker Error: ', response.error);
```
Comment (VitorDaynno) [inline src/components/profile.js:78]: Codacy Issue found: [Unexpected console statement.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/profile.js (hunk 14 above)
Comment (VitorDaynno) [inline src/components/profile.js:63]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/header.js (hunk 19 above)
Comment (VitorDaynno) [inline src/components/header.js:13]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

**Comment by VitorDaynno** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
- Added 6
           

Clones added
============
- src/pages/profile.js  1
- src/style/profile.js  2
- src/components/home.js  1
- src/pages/home.js  1
- src/components/profile.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182&bid=17780242)

File: src/components/header.js (hunk 3 above)
Comment (VitorDaynno) [inline src/components/header.js:18]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/profile.js (hunk 10 above)
Comment (VitorDaynno) [inline src/components/profile.js:45]: Codacy Issue found: [A function with a name starting with an uppercase letter should only be used as a constructor.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/profile.js (hunk 27 above)
Comment (VitorDaynno) [inline src/components/profile.js:76]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/profile.js (hunk 28 above)
Comment (VitorDaynno) [inline src/components/profile.js:78]: Codacy Issue found: [Unexpected console statement.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/profile.js (hunk 14 above)
Comment (VitorDaynno) [inline src/components/profile.js:63]: Codacy Issue found: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/header.js (hunk 19 above)
Comment (VitorDaynno) [inline src/components/header.js:13]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

**Comment by VitorDaynno** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones added
============
- src/components/home.js  1
- src/pages/profile.js  1
- src/style/profile.js  2
- src/components/login.js  2
- src/components/profile.js  2
- src/pages/home.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182&bid=17780242)

File: src/components/header.js [hunk 29]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  goBack = () => {
+    const {props} = this;
+    const {destiny} = props;
+    destiny();
+  };
+
+  render() {
+    const {name, isMenu} = this.props;
```
Comment (VitorDaynno) [inline src/components/header.js:19]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

File: src/components/header.js [hunk 30]
```diff
@@ -0,0 +1,37 @@
...
+
+import Styles from '../style/header';
+
+export default class Header extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  goBack = () => {
+    const {props} = this;
```
Comment (VitorDaynno) [inline src/components/header.js:13]: Codacy Issue found: [Unnecessary block.](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182)

**Comment by VitorDaynno** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/components/profile.js  2
- src/pages/profile.js  1
- src/components/home.js  1
- src/pages/home.js  1
- src/components/login.js  2
- src/style/profile.js  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182&bid=17780242)

**Comment by VitorDaynno** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/components/profile.js  2
- src/components/home.js  1
- src/pages/home.js  1
- src/style/profile.js  2
- src/components/login.js  1
- src/pages/profile.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Full-Moon-Castle/wizard/pullRequest?prid=5422182&bid=17780242)

**Review by cesarmacedo — APPROVED** [review]
LGTM

**Pull request merged**